src/script/conversation/ConversationRepository.ts

```diff
@@ -1522,16 +1532,30 @@ export class ConversationRepository {
   }
 
   /**
-   * Remove the current user from a Proteus conversation.
+   * Remove the current user from a conversation.
    *
    * @param conversationEntity Conversation to remove user from
    * @param clearContent Should we clear the conversation content from the database?
    * @returns Resolves when user was removed from the conversation
    */
-  private async leaveProteusConversation(conversationEntity: Conversation, clearContent: boolean) {
-    return clearContent
-      ? this.clearConversation(conversationEntity, true)
-      : this.removeMemberFromProteusConversation(conversationEntity, this.userState.self().qualifiedId);
+  private async leaveConversation(conversationEntity: Conversation, clearContent: boolean) {
+    if (clearContent) {
+      this.clearConversation(conversationEntity, false);
+    }
+
+    const userQualifiedId = this.userState.self().qualifiedId;
+
+    const roles = conversationEntity.roles();
+    delete roles[userQualifiedId.id];
+    conversationEntity.roles(roles);
+
+    const currentTimestamp = this.serverTimeHandler.toServerTimestamp();
+    const userLeaveEvent = await this.core.service!.conversation.leaveConversation(conversationEntity.qualifiedId);
+    const event =
+      userLeaveEvent || EventBuilder.buildMemberLeave(conversationEntity, userQualifiedId, true, currentTimestamp);
+
+    await this.eventRepository.injectEvent(event, EventRepository.SOURCE.BACKEND_RESPONSE);
+    return event;
```

Comment:
It's not clear to me why this change was made. The body is almost exactly similar to `removeMemberFromProteusConversation` why not just using it?

Reply:
Wanted to split it, because: `leaveConversation` is used for leaving both mls and proteus, so its generic function for leaving. Maybe I will extract duplicated code to separate, generic function, wdyt? :D

Reply:
are you sure? Both, `removeUserFromProteusConversation` and `leaveConversation` do the exact same API call 🤔  I don't see any difference between the 2 (except that `leaveConversation` uses the connected user Id)

Reply:
Oh, you're right, we have to name it more generic then and use one core method for 3 things: 
1. removing other user from proteus convo
2. leaving proteus convo
3. leaving mls convo
Does it sound good to you? :D

Reply:
That sounds perfect 👌